com.trollworks.gcs/src/com/trollworks/gcs/utility/Log.java

```diff
@@ -31,13 +31,16 @@
             } catch (Throwable throwable) {
                 error("Unable to redirect log to " + property, throwable);
             }
-        } else if (Platform.isWindows() && System.getenv() != null) {
-            File logDirectory = new File(System.getenv("LOCALAPPDATA") + "\\GCS");
-            try {
-                logDirectory.mkdirs();
-                OUT = new PrintStream(logDirectory+"\\error.log");
-            } catch (Throwable throwable) {
-                error("Unable to redirect log to " + logDirectory+"\\error.log", throwable);
+        } else if (Platform.isWindows()) {
+            String localAppData = System.getenv("LOCALAPPDATA");
+            if (localAppData != null) {
+                File logDirectory = new File(System.getenv("LOCALAPPDATA") + "\\GCS");
```

Comment:
Please use `localAppData` instead of calling `System.getenv()` a second time.

Reply:
That's embarrassing, I can't make an excuse for that one. See my amended commit.